pkg/events/events.go

```diff
@@ -0,0 +1,49 @@
+package events
+
+import (
+	"time"
+
+	svcsdk "github.com/aws/aws-sdk-go/service/memorydb"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	svcapitypes "github.com/aws-controllers-k8s/memorydb-controller/apis/v1alpha1"
+)
+
+const (
+	// The number of minutes worth of events to retrieve.
+	// 14 days
+	eventsDuration = time.Duration(336) * time.Hour
+)
+
+func NewDescribeEventsInput(
+	sourceName *string,
+	sourceType string,
```

Comment:
The SDK actually provides an enum for `SourceType`, you can find it in the constants section - https://docs.aws.amazon.com/sdk-for-go/api/service/memorydb/#pkg-constants

So this could be
```suggestion
sourceType svcsdk.SourceType,
```

and then you can pass values to it such as `svcsdk.SourceTypeACL`

Reply:
SourceType is not a constant or a field, so I don't find a thing as `svcsdk.SourceType`. This shows red as not defined. However, `svcsdk.SourceTypeAcl`, `SourceTypeCluster`, `SourceTypeUser` work.
https://pkg.go.dev/github.com/aws/aws-sdk-go/service/memorydb@v1.44.93
https://github.com/aws/aws-sdk-go/blob/v1.44.93/service/memorydb/api.go#L12248